Automattic/wp-calypso
# Store: Add promotions state

Disclaimer: While this looks like a big PR, around 400 lines of the code here is test code!

This adds action/reducer/handler code for promotions which are comprised
of "on-sale" products and coupons at this time. It fetches all product
and coupon data and compiles it into promotions objects.

To Test:
1. `npm test`
2. `npm start`
3. Visit `http://calypso.localhost:3000/store/promotions/<site url>`
4. Open redux dev tools.
5. Examine state at `extensions.woocommerce.sites.####.promotions`
6. Confirm that `products`, `coupons`, and `promotions` are filled out as per your site's data.
